### ISSUE TITLE
`$badfilter` rules are sensitive to domain lists

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/CoreLibs#1331]
Original issue:
https://github.com/AdguardTeam/AdguardBrowserExtension/issues/1690

related to https://github.com/AdguardTeam/CoreLibs/issues/1241 and https://github.com/AdguardTeam/AdguardKnowledgeBase/issues/469

**Comment by ameshkov** [comment]
This may be really useful

**Comment by sxgunchenko** [comment]
core/pull-requests/2100

**Comment by AdamWr** [comment]
I'm not sure if I should open new issue, but it seems that `$badfilter` doesn't work in some cases.

### Steps to reproduce
Add these rules:
```
||example.org^$replace=/<h1>[\s\S]*?<\/h1>/<iframe src="https:\/\/www.youtube.com\/embed\/dQw4w9WgXcQ"><\/iframe>/
||youtube.com/embed/
@@||youtube.com/embed/$domain=youtube.com|test.com
@@||youtube.com/embed/$domain=youtube.com|test.com,badfilter
```
Go here - http://example.org/

Exception rule should be disabled, but it seems that request is unblocked.

<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/29142494/95177915-10fb8880-07bf-11eb-8ba7-f578f7776e24.png)

</details>

AdGuard for Windows version 7.5 nightly 33 (build 3407, CL 1.7.114)

**Comment by sxgunchenko** [comment]
Looks like a bug
